core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -217,17 +200,40 @@ static Context deleteContext(Map<String, String> config) {
         Context dataContext = dataContext(config);
 
         String codecAsString = config.get(DELETE_AVRO_COMPRESSION);
-        CodecFactory codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
+        CodecFactory codec = codecAsString != null ? toCodec(codecAsString, config) : dataContext.codec();
 
         return new Context(codec);
       }
 
-      private static CodecFactory toCodec(String codecAsString) {
+      private static CodecFactory toCodec(String codecAsString, Map<String, String> config) {
+        CodecFactory codecFactory = null;
         try {
-          return CodecName.valueOf(codecAsString.toUpperCase(Locale.ENGLISH)).get();
+          switch (Codec.valueOf(codecAsString.toUpperCase(Locale.ENGLISH))) {
+            case UNCOMPRESSED:
+              codecFactory = CodecFactory.nullCodec();
+              break;
+            case SNAPPY:
+              codecFactory = CodecFactory.snappyCodec();
+              break;
+            case ZSTD:
+              codecFactory = CodecFactory.zstandardCodec(PropertyUtil.propertyAsInt(config,
+                  AVRO_ZSTD_COMPRESSION_LEVEL,
+                  AVRO_ZSTD_COMPRESSION_LEVEL_DEFAULT));
+              break;
+            case GZIP:
+              codecFactory = CodecFactory.deflateCodec(9);
+              break;
+            case LZ4:
+              break;
```

Comment:
I think LZ4 and BROTLI should not break but go to default?

Reply:
Ah yes, I had this fixed locally but didn't update my commit. The updated test will fail with an NPE when we set the codec if we don't fix this.